platform-controller/src/main/java/org/hobbit/controller/docker/ResourceInformationCollector.java

```diff
@@ -12,69 +17,92 @@
             containerMapping.put(c.id(), c);
         }
         ResourceUsageInformation resourceInfo = containerMapping.keySet().parallelStream()
                 // filter all containers that are not running
-                .filter(c -> TaskStatus.TASK_STATE_RUNNING.equals(containerMapping.get(c).state()))
+                .filter(c -> TaskStatus.TASK_STATE_RUNNING.equals(containerMapping.get(c).status().state()))
                 // get the stats for the single
                 .map(id -> requestCpuAndMemoryStats(id))
                 // sum up the stats
                 .collect(Collectors.reducing(ResourceUsageInformation::staticMerge)).orElse(null);
-        if (resourceInfo != null) {
-            // Add disk usage information
-            long diskUsage = containers.parallelStream().map(c -> c.sizeRootFs()).filter(s -> s != null)
-                    .mapToLong(s -> s).sum();
-            resourceInfo.setDiskStats(new DiskStats(diskUsage));
-        }
         return resourceInfo;
     }
 
     protected ResourceUsageInformation requestCpuAndMemoryStats(String containerId) {
-        ContainerStats stats = manager.getStats(containerId);
         ResourceUsageInformation resourceInfo = new ResourceUsageInformation();
-        if (stats != null) {
-            if ((stats.cpuStats() != null) && (stats.cpuStats().cpuUsage() != null)) {
-                resourceInfo.setCpuStats(new CpuStats(stats.cpuStats().cpuUsage().totalUsage()));
-            }
-            if ((stats.memoryStats() != null) && (stats.cpuStats().cpuUsage() != null)) {
-                resourceInfo.setMemoryStats(new MemoryStats(stats.memoryStats().usage()));
-            }
+        try {
+            resourceInfo.setCpuStats(new CpuStats(Math.round(Double.parseDouble(
+                    requestPrometheusValue(containerId, "container_cpu_usage_seconds_total"))*1000)));
+        } catch (Exception e) {
+            LOGGER.error("Could not get cpu usage stats for container {}", containerId, e);
+        }
+        try {
+            resourceInfo.setMemoryStats(new MemoryStats(Long.parseLong(
+                    requestPrometheusValue(containerId, "container_memory_usage_bytes"))));
+        } catch (Exception e) {
+            LOGGER.error("Could not get memory usage stats for container {}", containerId, e);
+        }
+        try {
+            resourceInfo.setDiskStats(new DiskStats(Long.parseLong(
+                    requestPrometheusValue(containerId, "container_fs_usage_bytes"))));
+        } catch (Exception e) {
+            LOGGER.error("Could not get disk usage stats for container {}", containerId, e);
         }
         return resourceInfo;
     }
 
+    private String requestPrometheusValue(String taskId, String metric) throws IOException, MalformedURLException {
+        String filter = "{container_label_com_docker_swarm_task_id=\"" + taskId + "\"}";
+        URL url = new URL("http://prometheus:9090/api/v1/query?query=" + metric + filter);
```

Comment:
prometheus host and port are hard coded, should be able to specify those with env var